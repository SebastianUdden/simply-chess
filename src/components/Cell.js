import React from "react"
import styled from "styled-components"
import { getColor, getChessPiece } from "../utils/chessPiece"

const Cell = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  min-width: 40px;
  min-height: 45px;
  background-color: ${p => p.color};
  color: #ccc;
  opacity: 0.9;
  :hover {
    cursor: pointer;
    opacity: 1;
    background-color: #bb8034;
  }
  :active {
    background-color: #794013;
  }
`

const Image = styled.img`
  width: 35px;
`

export default ({ row, cell, onClick }) => {
  const isSelected = cell.selected ? "isSelected" : ""
  return (
    <Cell
      id={cell.name + row.number}
      color={getColor(row, cell)}
      onClick={onClick}
      data-cy={isSelected}
    >
      {cell.symbol && (
        <Image src={getChessPiece(cell.symbol)} alt={cell.symbol.type} />
      )}
    </Cell>
  )
}
