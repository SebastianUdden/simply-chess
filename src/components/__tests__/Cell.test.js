import React from "react"
import renderer from "react-test-renderer"
import Cell from "../Cell"

describe("Cell", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Cell
          cell={{ name: "A", number: 1 }}
          row={{ number: 1 }}
          onClick={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
