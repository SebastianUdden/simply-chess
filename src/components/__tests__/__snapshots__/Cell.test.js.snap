// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell renders correctly 1`] = `
<div
  className="sc-AxjAm bbaRbW"
  color="#262626"
  data-cy=""
  id="A1"
  onClick={[Function]}
/>
`;
